### PR TITLE
Add auditing event identifiers for further FreeBSD syscalls:

### PR DESCRIPTION
**Pull request by jzck**
getcred(2)
setcred(2)
revertcred(2)

### CODE REVIEW

File: sys/bsm/audit_kevents.h [hunk 1]
```diff
@@ -652,6 +652,9 @@
 #define	AUE_GETUUID		43260	/* CADETS. */
 #define	AUE_LGETUUID		43261	/* CADETS. */
 #define	AUE_EXECVEAT		43262	/* FreeBSD/Linux. */
+#define AUE_GETCRED		43263	/* FreeBSD-specific. */
+#define AUE_SETCRED		43264	/* FreeBSD-specific. */
+#define AUE_REVERTCRED		43265	/* FreeBSD-specific. */
```
Comment (csjayp) [inline sys/bsm/audit_kevents.h]: Minor nit but can you use a tab instead of a space here?

**Comment by csjayp** [comment]
Hi, Have these syscalls been added to the kernel yet?

**Comment by jzck** [comment]
These syscalls have not yet been upstreamed, I'm doing this in preparation of an upcoming review. By looking at commits in both projects for other syscalls I figured this was the order in which to do it.
Also, if the review is ultimately not merged we can just revert this.

**Comment by jzck** [comment]
I'm upstreaming here to use the next free numbers in openbsm rather than freebsd which hasn't rebased in a while, as requested by @rwatson

**Comment by csjayp** [comment]
Sounds good, do you have a FreeBSD review open for these syscalls? I wouldn't mind getting more details on what they will be used for

**Comment by jzck** [comment]
I'm implementing the design here: https://wiki.freebsd.org/Per-Thread%20Credentials
I need to finish some tests before publishing, will posts links between the here and there

**Comment by jzck** [comment]
as was promised, here's the link to the FreeBSD review: https://reviews.freebsd.org/D18930

**Comment by csjayp** [comment]
What is the status of this work? I haven't seen movement on these syscalls for quite some time.

**Comment by cbrueffer** [comment]
This work seems to be sidelined for now (see https://reviews.freebsd.org/D18930), so I committed the additional PR in https://github.com/openbsm/openbsm/pull/54 first. Event numbers will need to be updated here.

**Comment by cemeyer** [comment]
I think we can close this out.  If any when someone implements something similar, it isn't too hard to look at this PR for insight, if that is useful.

(I had some concerns and suggestions about how to implement the functionality Jack/Gandi wanted in a more general way that were dismissed.  Which is fair; they had more limited needs.  But I think it's fairly plausible a future/better version of this work would require different syscalls added anyway.)

**Pull request closed without merging**